v2/js/components/homescreen.js

```diff
@@ -213,6 +217,72 @@ const HomeScreen = {
 			}
 		},
 
+		async getJournal() {
+			const response = await axios.get(`/api/v1/journal/${this.jid}`, {
+				headers: {
+					'x-key': this.token.x_key,
+					'x-access-token': this.token.access_token,
+				},
+			});
+
+			if (response.status != 200) {
+				throw new Error('Unable to load the journal');
+			}
+			if (response.status == 200) {
+				console.log(response.data);
+				const entries = response.data.entries;
+				const filteredEntries = entries.filter(entry => {
+					const activityId = entry.metadata.activity;
+					return this.activities.find(activity => activity.id === activityId);
+				}).map(entry => {
+					const activityId = entry.metadata.activity;
+					const matchingActivity = this.activities.find(activity => activity.id === activityId);
+
+					if (matchingActivity) {
+						const { ...restActivityData } = matchingActivity;
+						const dataToPush = {
+							...entry,
+							iconpath: restActivityData.directory + "/" + restActivityData.icon,
+						};
+						return dataToPush;
+					}
+				});
+
+				for (let i = 0; i < filteredEntries.length; i++) {
+					const entry = filteredEntries[i];
+					console.log(entry.metadata.activity)
+					const activityid = entry.metadata.activity;
+
+					setInterval(() => {
+						let iconRef = this.$refs["activity" + activityid][0];
+						if (iconRef !== undefined) {
+							iconRef.colorData = this.buddycolor;
+							clearInterval();
+						}
+					}, 1000);
```

Comment:
Hmmm. Not a good idea to wait.
Plus it's weird to update icons color in a method called `getJournal`.

Reply:
FYI, in v1 the color icons is decided during draw depending of the journal content (see https://github.com/llaske/sugarizer/blob/dev/js/homeview.js#L338). If the Journal has changed, the draw is relaunch.